app/services/energy_tariff_default_prices_creator.rb

```diff
@@ -5,10 +5,13 @@ def initialize(energy_tariff)
 
   def process
     return if @energy_tariff.flat_rate?
-    return if @energy_tariff.meters.empty?
+    return if @energy_tariff.tariff_holder_type == 'School' && @energy_tariff.meters.empty?
     return if @energy_tariff.energy_tariff_prices.any?
 
-    night_times = Economy7Times.times(@energy_tariff.meters.first.mpan_mprn)
+    night_times = case @energy_tariff.tariff_holder_type
+                  when 'School' then Economy7Times.times(@energy_tariff&.meters&.first&.mpan_mprn)
+                  when 'SiteSettings' then TimeOfDay.new(0, 0)..TimeOfDay.new(7, 0) # Equivalent to Economy7Times::DEFAULT_TIMES
```

Comment:
Analytics PR to make the private constant public https://github.com/Energy-Sparks/energy-sparks_analytics/pull/582
Will update the code here once the above is merged and tag released.